src/App.tsx

```diff
@@ -15,48 +16,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+const sorted = (goods: string[], sortBy: SortType, isReverse: boolean) => {
+  let prepearedGoods = [...goods];
+
+  if (sortBy) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+  const [goodsReverse, setGoodsReverse] = useState(false);
```

Comment:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```